### ISSUE TITLE
-[MGLMapView accessibilityElementAtIndex:] performs poorly on Simulator

### ISSUE THREAD

**Issue by friedbunny** [mapbox/mapbox-gl-native-ios#220]
When running XCUITests, `-[MGLMapView accessibilityElementAtIndex:]` can occasionally consume 100% of the Simulator’s CPU indefinitely. This can cause the tests to fail if the test harness was waiting for the app to idle before continuing.

Perhaps related to mapbox/mapbox-gl-native#13860 and/or mapbox/mapbox-gl-native-ios#172.

![Screen Shot 2019-04-03 at 1 31 20 PM](https://user-images.githubusercontent.com/1198851/55511264-fc13c700-5614-11e9-83cf-33a8d02ecef9.png)
_[see attached Instruments trace](https://github.com/mapbox/mapbox-gl-native/files/3040629/14324-Instruments.trace.zip)_

### Steps to reproduce

 Run XCUITests on Simulator, e.g., using `ios-sdk-examples` — intermittently pegs CPU.

### Configuration

**Mapbox SDK versions:** `ios-v4.10.0-beta.1`, others
**Device/simulator models:** iPhone SE (iOS 10.3.1), others
**Xcode version:** 10.2

/cc @julianrex @1ec5

**Comment by julianrex** [comment]
Transferred this issue to `mapbox-gl-native-ios` - this issue is still valid. We should look at continuing the work started in https://github.com/mapbox/mapbox-gl-native/pull/15303.

/cc @knov

**Comment by benjidea** [comment]
Hey, is there an update on this?

We (probably) have the same problem, but unfortunately on a real device. 
Once you enable voice over, the CPU is set to 100% and the app crashes when the memory reaches ~2GB.
The app is a bit complex but the ViewController with the MapView doesn't contain any additional elements.

A simple demo project (only one MapView) becomes unusable when Voice Over is activated for ~6 seconds, but does not crash.

**Comment by julianrex** [comment]
This issue was fixed by https://github.com/mapbox/mapbox-gl-native-ios/pull/318
